### DISCUSSION TITLE
Regarding patch decoding using rVAE

### DISCUSSION THREAD

**Discussion by utkarshp1161** [ziatdinovmax/pyroVED#56] [General]
Hi Max,

Richard(@RichardLiuCoding) and I tried below using regular vae and works great:

  1. m, s = vae.encode(patches)

  2. Choose a location in the latent space and then filter out patches
      with corresponding latent representation
      
      z = [-0.6,-0.4]
      
      Adjust this thres value to control how close these patches are w.r.t. the selected latent values
      thres = 0.25
      index = np.where(np.sqrt((m[:,0]-z[0])**2 + (m[:,1]-z[1])**2) < thres)
      
     
What I am trying now:

  1. trained a rvae: 
            rvae1 = pv.models.iVAE(in_dim, latent_dim=2,  # Number of latent dimensions other than the invariancies
                                  hidden_dim_e = [256, 256], # corresponds to the number of neurons in the hidden layers of the encoder
                                 hidden_dim_d = [256, 256],  # corresponds to the number of neurons in the hidden layers of the decoder
                                 invariances= ['r'], seed=0)
  
  2. Now I want to choose a z (randomly) and decode it through network

Question. since I have a rvae with 3 latent dim[z1 is angle and z2,z3 are other latent variables] shall I give a z with 3 enteries(I tried and it throws an error)? or two enteries(if two enteries what does these enteries correspond to?) Essentially I want to draw samples from the latent space or rvae and decode it and see.

**Comment by ziatdinovmax** [comment]
@utkarshp1161 you should be able to pass "angle" as a keyword argument: https://github.com/ziatdinovmax/pyroVED/blob/7807ffb1cb415b3cc76c1e02d52465a8ae0eeae4/pyroved/models/base.py#L159

**Reply by utkarshp1161** [reply]
@ziatdinovmax thanks a lot. We were able to use it.
